NDTensors/src/lib/Sectors/src/category_definitions/u1.jl

```diff
@@ -1,19 +1,25 @@
-using HalfIntegers: Half
-
 #
 # U₁ group (circle group, or particle number, total Sz etc.)
 #
 
-struct U1 <: AbstractCategory
-  n::Half{Int}
+# Parametric type to allow both integer label as well as
+# HalfInteger for easy conversion to/from SU(2)
+struct U1{T} <: AbstractCategory
+  n::T
 end
 
-dual(u::U1) = U1(-u.n)
+SymmetryStyle(::U1) = AbelianGroup()
+
+GradedAxes.dual(u::U1) = U1(-u.n)
 
-label(u::U1) = u.n
+category_label(u::U1) = u.n
 
-dimension(::U1) = 1
+trivial(::Type{U1}) = trivial(U1{Int})
+trivial(::Type{U1{T}}) where {T} = U1(T(0))
 
-trivial(::Type{U1}) = U1(0)
+label_fusion_rule(::Type{<:U1}, n1, n2) = n1 + n2
```

Comment:
I see that some categories define `label_fusion_rule` output a tuple of two outputs, a set of degeneracies and labels, while others just output a label (presumably the distinction is whether the groups are abelian or not).

I think that isn't a great generic code design, I think it is better to have a uniform output format, in which case even if it is trivial we should output both the degeneracies and labels for abelian groups, so here the output would be:
```julia
label_fusion_rule(::Type{<:U1}, n1, n2) = [1], [n1 + n2]
```
or maybe:
```julia
label_fusion_rule(::Type{<:U1}, n1, n2) = (1,), (n1 + n2,)
```
if you want to avoid allocating vectors unnecessarily, the main point is that it outputs two collections, where the first are the degeneracies and the second are the labels.

To get a bit fancier, the format could be that `label_fusion_rule` is expected to output a NamedTuple, i.e.:
```julia
label_fusion_rule(::Type{<:U1}, n1, n2) = (; degen=(1,), labels=(n1 + n2,))
```
personally I like that version since then the code that consumes the output of `label_fusion_rule` is nicer, since it allows accessing the fields of the NamedTuple with those names.

Reply:
@emstoudenmire curious to hear your thoughts on this one.

Reply:
One idea would be to define a `simple_label_fusion_rule` or `abelian_label_fusion_rule` that would be defined as:
```julia
simple_label_fusion_rule(::Type{<:U1}, n1, n2) = n1 + n2
```
and a generic function:
```julia
label_fusion_rule(C::Type{<:AbstractCategory}, n1, n2) = (; degen=(1,), labels=(simple_label_fusion_rule(C, n1, n2),))
```
(or maybe it could be a trait function based on `SymmetryStyle`). So then categories that need to will overload `label_fusion_rule`, but ones with simpler rules can just define `simple_label_fusion_rule`.

Reply:
I refactored `label_fusion_rule`. It now directly returns `degens, sectors`, as returning `labels` was a source of confusion.